src/transformers/utils/quantization_config.py

```diff
@@ -1079,3 +1079,64 @@ def get_loading_attributes(self):
         loading_attibutes = ["activation_scale_ub"]
         loading_attibutes_dict = {i: j for i, j in attibutes_dict.items() if i in loading_attibutes}
         return loading_attibutes_dict
+
+
+@dataclass
+class TorchAoConfig(QuantizationConfigMixin):
+    """This is a config class for torchao quantization/sparsity techniques.
+
+    Args:
+        quant_type (`str`):
+            The type of quantization we want to use, currently supporting: `int4_weight_only`, `int8_weight_only` and `int8_dynamic_activation_int8_weight`.
+        kwargs (`Dict[str, Any]`, *optional*):
+            The keyword arguments for the chosen type of quantization, for example, int4_weight_only quantization supports two keyword arguments
+            `group_size` and `inner_k_tiles` currently. More API examples and documentation of arguments can be found in
+            https://github.com/pytorch/ao/tree/main/torchao/quantization#other-available-quantization-techniques
+
+    Example:
+
+    ```python
+    quantization_config = TorchAoConfig("int4_weight_only", group_size=32)
+    # int4_weight_only quant is only working with *torch.bfloat16* dtype right now
+    model = AutoModelForCausalLM.from_pretrained(model_id, device_map="cuda", torch_dtype=torch.bfloat16, quantization_config=quantization_config)
+    ```
+    """
+
+    def __init__(self, quant_type: str, **kwargs):
+        self.quant_method = QuantizationMethod.TORCHAO
+        self.quant_type = quant_type
```

Comment:
Is there a way to specify which modules not to quantize just like how it is done in other quantization methods ? I think that this can be a great addition. For example, we skip the `lm_head` by default. Check the logic here: https://github.com/huggingface/transformers/blob/7e5d46ded433605a906fcab6be43ac85307cca9b/src/transformers/quantizers/quantizer_fbgemm_fp8.py#L170

Reply:
yeah we support this feature as well, I can add it

Reply:
we have a general `filter_fn` that takes a module and fully qualified name, and returns True if we want to quantize the module, I can expose that directly, does that sounds OK?

Reply:
Nice ! Whatever works the best for you ! I would advise you to just filter out the parameter you quantize to quantize or not in the `check_quantized_param` method.

Reply:
I see, `check_quantized_param` is already supported, but it's a global check that applies to all instances of quantizer so it does not allow finer grained control, I also had some difficulty adding `filter_fn` since it uses `torch` in the signature, I will change this to just accepting a list of fqns to skip for now

Reply:
I added the feature here: https://github.com/jerryzh168/transformers/pull/1 !